utils/s2n_blob.c

```diff
@@ -25,8 +25,11 @@
 
 bool s2n_blob_is_valid(const struct s2n_blob* b)
 {
-  bool blob_was_valid = S2N_OBJECT_PTR_IS_READABLE(b) && S2N_MEM_IS_READABLE(b->data,b->size);
-  return blob_was_valid;
+    return
+	b &&
+	S2N_OBJECT_PTR_IS_READABLE(b) &&
+	S2N_MEM_IS_READABLE(b->data,b->size) &&
```

Comment:
Do you also need to include S2N_MEM_IS_READABLE(b->data, b->allocated)?

Reply:
That's only true if growable is 1.

Reply:
I'm doing an experiment with a stronger checks in https://github.com/awslabs/s2n/pull/1667

Reply:
Updated